rotkehlchen/accounting/structures/base.py

```diff
@@ -277,6 +279,24 @@ def get_identifier(self) -> str:
     def get_assets(self) -> list[Asset]:
         return [self.asset]
 
+    def has_details(self) -> bool:
+        return (
+            self.extra_data is not None and
+            ('sub-swaps' in self.extra_data or 'liquity-staking' in self.extra_data)
+        )
```

Comment:
I don't see this scale easily. Why not return directly what is inside the extra_data field? This way you need to make sure always that whenever you add something that needs to be returned to also add it here but the only benefit that I see is having a common key but don't know that this can exists already by looking to the protocol in the frontend?